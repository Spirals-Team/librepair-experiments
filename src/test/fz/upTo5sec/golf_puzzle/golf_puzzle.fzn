array [1..2] of int: X_INTRODUCED_18 = [1,-1];
array [1..19] of int: X_INTRODUCED_40 = [1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1];
array [1..2] of int: X_INTRODUCED_45 = [-1,-1];
var -10..10: X_INTRODUCED_0;
var -10..10: X_INTRODUCED_1;
var -10..10: X_INTRODUCED_2;
var -10..10: X_INTRODUCED_3;
var -10..10: X_INTRODUCED_4;
var -10..10: X_INTRODUCED_5;
var -10..10: X_INTRODUCED_6;
var -10..10: X_INTRODUCED_7;
var -10..10: X_INTRODUCED_8;
var -10..10: X_INTRODUCED_9;
var -10..10: X_INTRODUCED_10;
var -10..10: X_INTRODUCED_11;
var -10..10: X_INTRODUCED_12;
var -10..10: X_INTRODUCED_13;
var -10..10: X_INTRODUCED_14;
var -10..10: X_INTRODUCED_15;
var -10..10: X_INTRODUCED_16;
var -10..10: X_INTRODUCED_17;
var 1..1000: club1:: output_var;
var 1..1000: club2:: output_var;
var 0..180: z:: output_var:: is_defined_var;
var 0..10: X_INTRODUCED_19 ::var_is_introduced ;
var 0..10: X_INTRODUCED_20 ::var_is_introduced ;
var 0..10: X_INTRODUCED_21 ::var_is_introduced ;
var 0..10: X_INTRODUCED_22 ::var_is_introduced ;
var 0..10: X_INTRODUCED_23 ::var_is_introduced ;
var 0..10: X_INTRODUCED_24 ::var_is_introduced ;
var 0..10: X_INTRODUCED_25 ::var_is_introduced ;
var 0..10: X_INTRODUCED_26 ::var_is_introduced ;
var 0..10: X_INTRODUCED_27 ::var_is_introduced ;
var 0..10: X_INTRODUCED_28 ::var_is_introduced ;
var 0..10: X_INTRODUCED_29 ::var_is_introduced ;
var 0..10: X_INTRODUCED_30 ::var_is_introduced ;
var 0..10: X_INTRODUCED_31 ::var_is_introduced ;
var 0..10: X_INTRODUCED_32 ::var_is_introduced ;
var 0..10: X_INTRODUCED_33 ::var_is_introduced ;
var 0..10: X_INTRODUCED_34 ::var_is_introduced ;
var 0..10: X_INTRODUCED_35 ::var_is_introduced ;
var 0..10: X_INTRODUCED_36 ::var_is_introduced ;
var -10000..10000: X_INTRODUCED_41 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_42 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_46 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_47 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_49 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_50 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_52 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_53 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_55 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_56 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_58 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_59 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_61 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_62 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_64 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_65 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_67 ::var_is_introduced :: is_defined_var;
var -10000..10000: X_INTRODUCED_68 ::var_is_introduced :: is_defined_var;
array [1..18] of var int: holes:: output_array([1..9,1..2]) = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_8,X_INTRODUCED_9,X_INTRODUCED_10,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_13,X_INTRODUCED_14,X_INTRODUCED_15,X_INTRODUCED_16,X_INTRODUCED_17];
array [1..19] of var int: X_INTRODUCED_39 ::var_is_introduced  = [z,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_21,X_INTRODUCED_22,X_INTRODUCED_23,X_INTRODUCED_24,X_INTRODUCED_25,X_INTRODUCED_26,X_INTRODUCED_27,X_INTRODUCED_28,X_INTRODUCED_29,X_INTRODUCED_30,X_INTRODUCED_31,X_INTRODUCED_32,X_INTRODUCED_33,X_INTRODUCED_34,X_INTRODUCED_35,X_INTRODUCED_36];
constraint int_lin_le(X_INTRODUCED_18,[club1,club2],0);
constraint int_abs(X_INTRODUCED_0,X_INTRODUCED_19);
constraint int_abs(X_INTRODUCED_1,X_INTRODUCED_20);
constraint int_abs(X_INTRODUCED_2,X_INTRODUCED_21);
constraint int_abs(X_INTRODUCED_3,X_INTRODUCED_22);
constraint int_abs(X_INTRODUCED_4,X_INTRODUCED_23);
constraint int_abs(X_INTRODUCED_5,X_INTRODUCED_24);
constraint int_abs(X_INTRODUCED_6,X_INTRODUCED_25);
constraint int_abs(X_INTRODUCED_7,X_INTRODUCED_26);
constraint int_abs(X_INTRODUCED_8,X_INTRODUCED_27);
constraint int_abs(X_INTRODUCED_9,X_INTRODUCED_28);
constraint int_abs(X_INTRODUCED_10,X_INTRODUCED_29);
constraint int_abs(X_INTRODUCED_11,X_INTRODUCED_30);
constraint int_abs(X_INTRODUCED_12,X_INTRODUCED_31);
constraint int_abs(X_INTRODUCED_13,X_INTRODUCED_32);
constraint int_abs(X_INTRODUCED_14,X_INTRODUCED_33);
constraint int_abs(X_INTRODUCED_15,X_INTRODUCED_34);
constraint int_abs(X_INTRODUCED_16,X_INTRODUCED_35);
constraint int_abs(X_INTRODUCED_17,X_INTRODUCED_36);
constraint int_lin_eq(X_INTRODUCED_40,X_INTRODUCED_39,0):: defines_var(z);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_42,X_INTRODUCED_41],-150);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_47,X_INTRODUCED_46],-300);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_50,X_INTRODUCED_49],-250);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_53,X_INTRODUCED_52],-325);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_56,X_INTRODUCED_55],-275);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_59,X_INTRODUCED_58],-350);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_62,X_INTRODUCED_61],-225);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_65,X_INTRODUCED_64],-400);
constraint int_lin_eq(X_INTRODUCED_45,[X_INTRODUCED_68,X_INTRODUCED_67],-425);
constraint int_times(X_INTRODUCED_1,club2,X_INTRODUCED_41):: defines_var(X_INTRODUCED_41);
constraint int_times(X_INTRODUCED_0,club1,X_INTRODUCED_42):: defines_var(X_INTRODUCED_42);
constraint int_times(X_INTRODUCED_3,club2,X_INTRODUCED_46):: defines_var(X_INTRODUCED_46);
constraint int_times(X_INTRODUCED_2,club1,X_INTRODUCED_47):: defines_var(X_INTRODUCED_47);
constraint int_times(X_INTRODUCED_5,club2,X_INTRODUCED_49):: defines_var(X_INTRODUCED_49);
constraint int_times(X_INTRODUCED_4,club1,X_INTRODUCED_50):: defines_var(X_INTRODUCED_50);
constraint int_times(X_INTRODUCED_7,club2,X_INTRODUCED_52):: defines_var(X_INTRODUCED_52);
constraint int_times(X_INTRODUCED_6,club1,X_INTRODUCED_53):: defines_var(X_INTRODUCED_53);
constraint int_times(X_INTRODUCED_9,club2,X_INTRODUCED_55):: defines_var(X_INTRODUCED_55);
constraint int_times(X_INTRODUCED_8,club1,X_INTRODUCED_56):: defines_var(X_INTRODUCED_56);
constraint int_times(X_INTRODUCED_11,club2,X_INTRODUCED_58):: defines_var(X_INTRODUCED_58);
constraint int_times(X_INTRODUCED_10,club1,X_INTRODUCED_59):: defines_var(X_INTRODUCED_59);
constraint int_times(X_INTRODUCED_13,club2,X_INTRODUCED_61):: defines_var(X_INTRODUCED_61);
constraint int_times(X_INTRODUCED_12,club1,X_INTRODUCED_62):: defines_var(X_INTRODUCED_62);
constraint int_times(X_INTRODUCED_15,club2,X_INTRODUCED_64):: defines_var(X_INTRODUCED_64);
constraint int_times(X_INTRODUCED_14,club1,X_INTRODUCED_65):: defines_var(X_INTRODUCED_65);
constraint int_times(X_INTRODUCED_17,club2,X_INTRODUCED_67):: defines_var(X_INTRODUCED_67);
constraint int_times(X_INTRODUCED_16,club1,X_INTRODUCED_68):: defines_var(X_INTRODUCED_68);
solve  minimize z;
