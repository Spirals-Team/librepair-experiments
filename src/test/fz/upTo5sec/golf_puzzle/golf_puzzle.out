club1 = 125;
club2 = 150;
z = 26;
holes = array2d(1..9, 1..2, [0, 1, 0, 2, 2, 0, -1, 3, 1, 1, 4, -1, 3, -1, 2, 1, 1, 2]);
----------
==========